### PR TITLE
Improve N3LO Pgg (large N), Pqg and Pqq

### PR DESCRIPTION
**Pull request by giacomomagni**
This PR is to introduce some update on N3LO splitting functions:  

* an additional constrain on the large-N limit of $P_{gg}^{(3)}$ which has been overlooked, see https://arxiv.org/abs/hep-ph/0511302.  In particular we fix the coefficient $C_4$ in: 

   $\gamma_{gg}^{(3)} \approx A_4 S_1(N) + B_4 + C_4 \frac{S_1(N)}{N}$

  which was previously fitted.

  ```
  c4_theory = 85814.1 - 13880.5 nf + 135.111 nf^2
  c4_fitted = 110990. - 15528.6 nf + 231.044 nf^2
  ```

  Introduced changes should be within uncertainties, although visible.
  The updated splitting function is: 
  
  [gamma_gg_msht_ratio_new.pdf](https://github.com/NNPDF/eko/files/11879862/gamma_gg_msht_ratio_new.pdf)
  [gamma_gg_msht_logx_new.pdf](https://github.com/NNPDF/eko/files/11879861/gamma_gg_msht_logx_new.pdf)
  
  to be compared to the one in master:
  
  [gamma_gg_msht_ratio.pdf](https://github.com/NNPDF/eko/files/11879856/gamma_gg_msht_ratio.pdf)
  [gamma_gg_msht_logx.pdf](https://github.com/NNPDF/eko/files/11879858/gamma_gg_msht_logx.pdf)

* Update the determination of $P_{qg}$ according to the new moments published in https://arxiv.org/pdf/2307.04158.pdf
   This change has a major impact on IHOU.

* Slightly simplify $P_{qq, PS}$ determination.

### CODE REVIEW

File: doc/source/theory/N3LO_ad.rst [hunk 1]
```diff
@@ -362,13 +363,13 @@ and following :cite:`Falcioni:2023luc` we use:
         :align: center
 
         *   - :math:`f_1(N)`
-            - :math:`\frac{1}{(N-1)^2}`
+            - :math:`\frac{1}{(N-1)^2} + \frac{1}{N^2}`
```
Comment (felixhekhorn) [inline doc/source/theory/N3LO_ad.rst:366]: `qq` is unchanged here, right? so this is a typo?

File: doc/source/theory/N3LO_ad.rst [hunk 2]
```diff
@@ -229,10 +229,11 @@ The other parts are approximated using some known limits:
         It is known that :cite:`Albino:2000cp,Moch:2021qrk` the diagonal terms diverge in N-space as:
 
             .. math ::
-                \gamma_{kk} \approx A^{(r)}_4 S_1(N) + B^{(r)}_4 + \mathcal{O}(1)
+                \gamma_{kk} \approx A^{(r)}_4 S_1(N) + B^{(r)}_4 + C^{(r)}_4 \frac{S_1(N)}{N} + \mathcal{O}(\frac{1}{N})
 
         Where again the coefficient :math:`A^{(r)}_4` is the |QCD| cusp anomalous dimension for the adjoint or fundamental representation.
         Again the coefficient :math:`B^{(r)}_4` has been extracted from soft anomalous dimensions :cite:`Duhr:2022cob`.
+        While :math:`C^{(r)}_4`can be estimate from lower order :cite:`Dokshitzer:2005bf`.
```
Comment (felixhekhorn) [inline doc/source/theory/N3LO_ad.rst:236]:
```suggestion
        While :math:`C^{(r)}_4`can be estimate from lower orders :cite:`Dokshitzer:2005bf`.
```
maybe you can connect the three sentences? :upside_down_face:

File: doc/source/theory/N3LO_ad.rst (hunk 1 above)
Comment (giacomomagni) [reply doc/source/theory/N3LO_ad.rst:366]: this was a typo indeed

**Review by felixhekhorn — APPROVED** [review]

File: doc/source/theory/N3LO_ad.rst [hunk 3]
```diff
@@ -339,38 +339,33 @@ final reduced sets of candidates.
...
         :align: center
 
         *   - :math:`f_1(N)`
-            - :math:`\frac{1}{(N-1)^2} + \frac{1}{N^2}`
+            - :math:`-\frac{1}{(N-1)^2} + \frac{1}{N^2}`
         *   - :math:`f_2(N)`
             - :math:`\mathcal{M}[(1-x)\ln^2(1-x)]`
         *   - :math:`f_{3,\dots,8}(N)`
-            - :math:`\frac{1}{N^4},\ \frac{1}{N^3},\ \mathcal{M}[(1-x)\ln(1-x)],\ \mathcal{M}[(1-x)^2\ln^2(1-x)],\ \mathcal{M}[(1-x)^2\ln(1-x)],\ \frac{1}{N-1}-\frac{1}{N}`
+            - :math:`\frac{1}{N^4},\ \frac{1}{N^3},\ \mathcal{M}[(1-x)\ln(1-x)],\ \mathcal{M}[(1-x)^2\ln(1-x)^2],\ \frac{1}{N-1}-\frac{1}{N},\ \mathcal{M}[(1-x)\ln(x)]`
```
Comment (felixhekhorn) [inline doc/source/theory/N3LO_ad.rst:366]:
```suggestion
            - :math:`\frac{1}{N^4},\ \frac{1}{N^3},\ \mathcal{M}[(1-x)\ln(1-x)],\ \mathcal{M}[(1-x)^2\ln^2(1-x)],\ \frac{1}{N-1}-\frac{1}{N},\ \mathcal{M}[(1-x)\ln(x)]`
```
this is more consistent and clearer

**Review by felixhekhorn — APPROVED** [review]

**Pull request merged**